d2core/d2ui/d2ui.go

```diff
@@ -17,122 +19,105 @@ const (
 	CursorButtonRight CursorButton = 2
 )
 
-var widgets []Widget
-var cursorButtons CursorButton
-var pressedIndex int
-var CursorX int
-var CursorY int
+type UI struct {
+	widgets       []Widget
+	cursorButtons CursorButton // TODO (carrelld) convert dependent code and remove
+	CursorX       int          // TODO (carrelld) convert dependent code and remove
+	CursorY       int          // TODO (carrelld) convert dependent code and remove
+	pressedWidget Widget
+}
+
+var singleton UI
 var clickSfx d2audio.SoundEffect
-var waitForLeftMouseUp bool
 
 func Initialize() {
-	pressedIndex = -1
-	clickSfx, _ = d2audio.LoadSoundEffect(d2resource.SFXButtonClick)
-	waitForLeftMouseUp = false
+	sfx, err := d2audio.LoadSoundEffect(d2resource.SFXButtonClick)
+	if err != nil {
+		log.Fatalf("failed to initialize ui: %v", err)
+	}
+	clickSfx = sfx
+
+	d2input.BindHandler(&singleton) // TODO bind here or at the calling code
 }
 
 // Reset resets the state of the UI manager. Typically called for new screens
 func Reset() {
-	widgets = make([]Widget, 0)
-	pressedIndex = -1
-	waitForLeftMouseUp = true
+	singleton = UI{}
```

Comment:
Given that we BindHandler on the pointer to the singleton above, we should probably reset fields (like the original code did) or UnbindHandler and then BindHandler as part of this Reset method